includes/subscriptions/class-wc-payments-product-service.php

```diff
@@ -128,11 +128,22 @@ public function get_wcpay_product_id( WC_Product $product, $test_mode = null ) :
 	 * return string       The item's WCPay product id.
 	 */
 	public function get_wcpay_product_id_for_item( string $type ) : string {
-		if ( ! get_option( self::get_wcpay_product_id_option() . '_' . $type ) ) {
-			$this->create_product_for_item_type( $type );
+		$sanitized_type  = self::sanitize_option_key( $type );
+		$option_key_name = self::get_wcpay_product_id_option() . '_' . $sanitized_type;
+		if ( ! get_option( $option_key_name ) ) {
+			$this->create_product_for_item_type( $sanitized_type );
 		}
+		return get_option( $option_key_name );
+	}
 
-		return get_option( self::get_wcpay_product_id_option() . '_' . $type );
+	/**
+	 * Sanitize option key string to replace space with underscore, and remove special characters.
+	 *
+	 * @param string $type Non sanitized input.
+	 * return string       Sanitized output.
+	 */
+	public static function sanitize_option_key( string $type ) {
+		return sanitize_key( str_replace( ' ', '_', strtolower( trim( $type ) ) ) );
```

Comment:
`sanitize_key` will also [apply strtolower](https://github.com/WordPress/wordpress-develop/blob/6.0/src/wp-includes/formatting.php#L2158). We can remove our explicit use of it.

Reply:
Thanks for sharing the reference! I will remove the `strtolower` from my line, since it is already taken care of by `sanitize_key`

Reply:
Fixed here https://github.com/Automattic/woocommerce-payments/pull/4496/commits/9c3f75c0d55c429575811709930442123a648c34